pkg/datastore/database.go

```diff
@@ -315,12 +315,59 @@ func (db *relationalDb) InsertInTable(ctx context.Context, tableName string, rec
 }
 
 /*
-Deletes a record from a DB table.
+Deletes a record from a DB table, soft deletes if `DeletedAt` attribute is present.
+*/
+func (db *relationalDb) SoftDelete(ctx context.Context, record Record) (rowsAffected int64, err error) {
+	return db.SoftDeleteInTable(ctx, GetTableName(record), record)
+}
+
+/*
+Deletes a record from a DB table. (to be used after soft-delete for structs with `DeletedAt` field).
 */
 func (db *relationalDb) Delete(ctx context.Context, record Record) (rowsAffected int64, err error) {
 	return db.DeleteInTable(ctx, GetTableName(record), record)
 }
 
+/*
+Deletes a record from DB table tableName, soft deletes if `DeletedAt` attribute is present.
+*/
+func (db *relationalDb) SoftDeleteInTable(ctx context.Context, tableName string, record Record) (rowsAffected int64, err error) {
+	tx, err := db.GetDBTransaction(ctx, tableName, record)
+	if err != nil {
+		return 0, err
+	}
+
+	// Execute query
+	tx.Delete(record)
+	tx.Commit()
+	if tx.Error != nil {
+		err = ErrExecutingSqlStmt.Wrap(tx.Error)
+		db.logger.Errorln(err)
+		return 0, err
+	}
+	return tx.RowsAffected, nil
+}
+
+/*
+Deletes a record from DB table tableName. (to be used after soft-delete for structs with 'DeletedAt` fields).
+*/
+func (db *relationalDb) DeleteInTable(ctx context.Context, tableName string, record Record) (rowsAffected int64, err error) {
+	tx, err := db.GetDBTransaction(ctx, tableName, record)
+	if err != nil {
+		return 0, err
+	}
+
+	// Execute query
+	tx.Unscoped().Delete(record)
```

Comment:
Shouldn't you check if `Delete` failed (by checking if `tx.Error` is not nil) before committing? (I see that everywhere in the code you're not checking `tx.Error` before committing)

Reply:
done